packages/dht/src/connection/WebSocket/WebSocketConnector.ts

```diff
@@ -0,0 +1,198 @@
+        fnCanConnect: (peerDescriptor: PeerDescriptor, _ip: string, port: number) => boolean
+    ) {
+        super()
+        this.canConnectFunction = fnCanConnect.bind(this)
+
+        this.rpcCommunicator = new RoutingRpcCommunicator(this.WESOCKET_CONNECTOR_APP_ID, this.rpcTransport, {
+            rpcRequestTimeout: 10000
+        })
+
+        this.requestConnection = this.requestConnection.bind(this)
+
+        this.rpcCommunicator.registerRpcMethod(
+            WebSocketConnectionRequest,
+            WebSocketConnectionResponse,
+            'requestConnection',
+            this.requestConnection
+        )
+    }
+
+    connect({ host, port, url, ownPeerDescriptor, targetPeerDescriptor }: {
+        host?: string,
+        port?: number,
+        url?: string,
+        ownPeerDescriptor?: PeerDescriptor,
+        targetPeerDescriptor?: PeerDescriptor
+    } = {}
+    ): IConnection {
+
+        if (!host && !port && !url && ownPeerDescriptor && targetPeerDescriptor) {
+            return this.requestConnectionFromPeer(ownPeerDescriptor, targetPeerDescriptor)
+        }
+        const socket = new ClientWebSocket()
+
+        socket.once(ConnectionEvent.CONNECTED, () => {
+            this.emit(ConnectionSourceEvent.CONNECTED, socket)
+        })
+
+        let address = ''
+        if (url) {
+            address = url
+        }
+        else if (host && port) {
+            address = 'ws://' + host + ':' + port
+        }
+
+        socket.connect(address)
+        return socket
+    }
+
+    connectAsync({ host, port, url, timeoutMs }:
+        { host?: string; port?: number; url?: string; timeoutMs: number } = { timeoutMs: 1000 }): Promise<IConnection> {
+
+        return new Promise((resolve, reject) => {
+            const socket = new ClientWebSocket()
+
+            const connectHandler = () => {
+                clearTimeout(timeout)
+                socket.off(ConnectionEvent.ERROR, errorHandler)
+                resolve(socket)
+            }
+
+            const errorHandler = () => {
+                //console.log('errorHandler of WebSocketConnector::connectAsync()')
+                clearTimeout(timeout)
+                reject()
+            }
+
+            const timeoutHandler = () => {
+                socket.off(ConnectionEvent.ERROR, errorHandler)
+                reject()
+            }
+
+            const timeout = setTimeout(timeoutHandler, timeoutMs)
+
+            socket.once(ConnectionEvent.CONNECTED, connectHandler)
+            socket.once(ConnectionEvent.ERROR, errorHandler)
+
+            let address = ''
+            if (url) {
+                address = url
+            }
+            else if (host && port) {
+                address = 'ws://' + host + ':' + port
+            }
+
+            socket.connect(address)
+        })
+    }
+
+    // Security check
+    withinPortRange(port: number): boolean {
+        // Check that requested connections is withing acceted range
+        return !!port
```

Comment:
Currently it does not check anything but eventually we will have to limit the port range that can be used for the WebSocket servers. This is strictly a security measure to avoid certain types of attacks.

Reply:
OTOH, no need to keep placeholders, if we don't plan to implement soon

Reply:
> What is the purpose of this check? Currently it returns true for every other number except `0`.
> 
> We could consider avoiding `!! `operator in all codebase as it is usually very confusing. If we need e.g. a check about nulliness somewhere, `foo !== undefined` (or `foo != null`) is significantly more readable and explicit.

@teogeb @juslesan @harbu 
I do not agree with this. In javascript 'falsy' as expressed by operator ! or !! covers a lot more cases than simple null or undefined, and using these explicitly takes a lot of space and is error-prone (it easy to forget one). Here is a full list of the cases that ! or !! covers:  

console.log(isFalsy(null));
console.log(isFalsy(undefined));
console.log(isFalsy(NaN));
console.log(isFalsy(0));
console.log(isFalsy(false));
console.log(isFalsy(''));
console.log(isFalsy([]));
console.log(isFalsy({}));